src/solve.jl

```diff
@@ -6,6 +6,15 @@ function DiffEqBase.__solve(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase
   integrator.sol
 end
 
+"""
+    DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.AbstractDAEProblem}, args...; kwargs...)
```

Comment:
no. Just do this at the DiffEqBase level. This will just make things harder to maintain.